catkin_home/src/main_engine/src/main_engine/Action.py

```diff
@@ -0,0 +1,114 @@
+import rospy
+import actionlib
+import time
+import sys
+import actions.msg
+import rb_home_arm.msg
+# TODO: Import all the messages related to actions I guess ?
+'''
+All Goals, and Actions must be imported here from he corresponding packages
+'''
+
+
+class Action(object):
+    def __init__(self, id, action_client_binded, args):
+        self.id = id
+        self.args = args
+        # Possible States Are: PENDING, ACTIVE, RECALLED, REJECTED, PREEMPTED, ABORTED, SUCCEEDED, LOST.
+        self.feedback = ["PENDING", "ACTIVE", "RECALLED",
+                         "SUCCEEDED", "PREEMPTED", "ABORTED", "REJECTED", "LOST"]
+        self.action_client_name = action_client_binded
+        self.action_client = None
+        self.time_requested = int(time.time())
+        self.specific_function = {
+            "bring_something": self.bring_something,
+            "go_to": self.go_to,
+            "pick_up":self.pick_up,
+            "put_down":self.put_down,
+            "default": self.specific_function_not_found,
+        }
+        self.specific_function.setdefault("default")
+
+    def specific_function_not_found(self):
+        print("Not found the corresponding action")
+
+    def print_self(self, endline=False):
+        if(endline == False):
+            print(self.get_description(),)
+            return
+        print(self.get_description())
```

Comment:
what's the point of the if?

Reply:
If i want to print them all in the same line I call the function without endline.